pydocusign/client.py

```diff
@@ -450,12 +450,12 @@ def post_recipient_view(self, authenticationMethod=None,
         data = {
             'authenticationMethod': authenticationMethod,
             'clientUserId': clientUserId,
-            'email': email,
-            'envelopeId': envelopeId,
             'returnUrl': returnUrl,
-            'userId': userId,
-            'userName': userName,
         }
+        for key in ('email', 'userName', 'userId'):
+            # ignore empty values; they won't help with a lookup
+            if locals()[key]:
+                data[key] = locals()[key]
```

Comment:
@ewjoachim yeah, i don't love using `locals` but also wasn't satisfied with any other option i could think of which would avoid changing the function signature...